gcc/config/riscv/riscv-modes.def

```diff
@@ -20,3 +20,216 @@ along with GCC; see the file COPYING3.  If not see
 <http://www.gnu.org/licenses/>.  */
 
 FLOAT_MODE (TF, 16, ieee_quad_format);
+
+/* Half-precision floating point for __fp16.  */
+FLOAT_MODE (HF, 2, 0);
+ADJUST_FLOAT_FORMAT (HF, &ieee_half_format);
```

Comment:
Yearh, you are right. I did missed something in some commits and make some commits mixed up. Sorry for that. Since the GCC was developed and it's painful to split them into several commits. Maybe in the next 1 ~ 2 weeks, after I finish the C++ overloaded functions. I am gonna split a clean GCC that only support RVV intrinsic (No any middle-end change) and test it fully. Then I make a new PR for it, at that time I will try my best to split them more reasonably for you to review (for example: zfh support, register spill.....etc). Before that, can you review the GCC codes base on this PR for the moment?  I am glad that you gave me suggestions.